File: packages/js-sdk/src/regex/detectors/email.ts
// Email address detector

import { RegexDetector } from '../base'
import { EntityType } from '../entities'
import { registerUniversal } from '../registry'

class EmailDetector extends RegexDetector {
  entityType = EntityType.EMAIL_ADDRESS
  score = 0.95
  pattern = /\b[a-zA-Z0-9._%+\-]+@[a-zA-Z0-9.\-]+\.[a-zA-Z]{2,}\b/g

  preCheck(text: string): boolean {
    return text.includes('@')
  }
}

registerUniversal(EmailDetector)

export { EmailDetector }

File: packages/python-sdk/src/blindfold/regex/detectors/email.py
"""Email address detector"""

import re

from ..base import RegexDetector
from ..entities import EntityType
from ..registry import register_universal


@register_universal
class EmailDetector(RegexDetector):
    entity_type = EntityType.EMAIL_ADDRESS
    score = 0.95
    needs_digit = False
    pattern = re.compile(
        r"\b[a-zA-Z0-9._%+\-]+@[a-zA-Z0-9.\-]+\.[a-zA-Z]{2,}\b"
    )

    def pre_check(self, text: str) -> bool:
        return "@" in text
